src/controllers/executionPlanWebviewController.ts

```diff
@@ -0,0 +1,168 @@
+          context.extensionUri,
+          "media",
+          "executionPlan.svg"
+        ),
+      }
+    );
+    this.state.isLoading = true;
+    this.initialize();
+  }
+
+  private async initialize() {
+    this.state.sqlPlanContent = this.executionPlanContents;
+	  this.state.theme = vscode.window.activeColorTheme.kind === vscode.ColorThemeKind.Dark ? "dark" : "light";
+    this.state.localizedConstants = LocalizedConstants;
+
+    await this.getExecutionPlan();
+    this.state.totalCost = this.calculateTotalCost();
+    this.registerRpcHandlers();
+    this.state.isLoading = false;
+  }
+
+  private registerRpcHandlers() {
+    this.registerReducer("getExecutionPlan", async (state, payload) => {
+      await this.getExecutionPlan();
+
+      return {
+        ...state,
+        executionPlan: this.state.executionPlan,
+        executionPlanGraphs: this.state.executionPlanGraphs
+      };
+    });
+    this.registerReducer("saveExecutionPlan", async (state, payload) => {
+      const homeDir = require("os").homedir(); // Get the user's home directory
+      const documentsFolder = vscode.Uri.file(`${homeDir}/Documents`);
+
+      let filename: vscode.Uri;
+      let counter = 0;
+      if (await this.fileExists(documentsFolder, `plan.sqlplan`)) {
+        counter += 1;
+        while (
+          await this.fileExists(documentsFolder, `plan${counter}.sqlplan`)
+        ) {
+          counter += 1;
+        }
+        filename = vscode.Uri.joinPath(
+          documentsFolder,
+          `plan${counter}.sqlplan`
+        );
+      } else {
+        filename = vscode.Uri.joinPath(documentsFolder, "plan.sqlplan");
+      }
+
+      // Show a save dialog to the user
+      const saveUri = await vscode.window.showSaveDialog({
+        defaultUri: filename,
+        filters: {
+          "SQL Plan Files": ["sqlplan"],
+        },
+      });
+
+      if (saveUri) {
+        // Write the content to the new file
+        await vscode.workspace.fs.writeFile(
+          saveUri,
+          Buffer.from(payload.sqlPlanContent)
+        );
+      }
+
+      return state;
+    });
+    this.registerReducer("showPlanXml", async (state, payload) => {
+      const planXmlDoc = await vscode.workspace.openTextDocument({
+        content: payload.sqlPlanContent,
+        language: 'xml'
+      });
+
+      await vscode.window.showTextDocument(planXmlDoc);
+
+      return state;
+    });
+    this.registerReducer("showQuery", async (state, payload) => {
+      const sqlDoc = await vscode.workspace.openTextDocument({
+        content: payload.query,
+        language: 'sql'
+      });
+
+      await vscode.window.showTextDocument(sqlDoc);
+
+      return state;
+    });
+    this.registerReducer("updateTotalCost", async (state, payload) => {
+      this.state.totalCost += payload.totalCost;
+
+      return {
+        ...state,
+        totalCost: this.state.totalCost
+      };
+    });
+  }
+
+  private async getExecutionPlan() {

```

Comment:
typically "get" methods return a value, maybe createExecutionPlanGraphs or something would better describe what's happening in the method.

Reply:
changed it!